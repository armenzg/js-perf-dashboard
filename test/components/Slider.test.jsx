import React from 'react';
import renderer from 'react-test-renderer';
import 'raf/polyfill';
import Slider from '../../src/components/Slider';

it('renders correctly', () => {
  const tree = renderer
    .create((
      <Slider
        identifier="test"
        label="Test"
        searchParam="test"
        selectedValue={15}
        options={{ min: 10, max: 30, step: 5 }}
        onChangeUpdateTooltipFunc={() => ('test')}
        handleSliderChange={() => {}}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
