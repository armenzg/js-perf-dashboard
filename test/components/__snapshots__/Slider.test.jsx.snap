// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  autoComplete="off"
  className="Slider-root-1"
>
  <label
    className="MuiFormLabel-root-10 MuiInputLabel-root-3 MuiInputLabel-animated-7 Slider-label-2"
    htmlFor="test"
  >
    Test
  </label>
  <div
    aria-describedby={null}
    aria-labelledby="test"
    aria-orientation="horizontal"
    aria-valuemax={30}
    aria-valuemin={10}
    aria-valuenow={15}
    className="MuiSlider-root-24"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    onTouchStartCapture={[Function]}
    role="slider"
    title="test"
  >
    <div
      className="MuiSlider-container-25"
    >
      <div
        className="MuiSlider-track-26 MuiSlider-trackBefore-27"
        style={
          Object {
            "width": "25%",
          }
        }
      />
      <button
        className="MuiButtonBase-root-37 MuiSlider-thumb-29"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onTouchStartCapture={[Function]}
        style={
          Object {
            "left": "25%",
          }
        }
        tabIndex="0"
        type="button"
      />
      <div
        className="MuiSlider-track-26 MuiSlider-trackAfter-28"
        style={
          Object {
            "width": "calc(100% - 5px)",
          }
        }
      />
    </div>
  </div>
</form>
`;
